pyteal/ast/abi/array_dynamic.py

```diff
@@ -102,3 +97,35 @@ def length(self) -> Expr:
 
 
 DynamicArray.__module__ = "pyteal.abi"
+
+
+class DynamicBytes(DynamicArray[Byte]):
+    """The convenience class that represents ABI static byte array."""
+
+    def __init__(self) -> None:
+        super().__init__(DynamicArrayTypeSpec(ByteTypeSpec()))
+
+    def set(
+        self,
+        values: Union[
+            bytes,
+            bytearray,
+            Expr,
+            Sequence[Byte],
+            DynamicArray[Byte],
+            ComputedValue[DynamicArray[Byte]],
+        ],
```

Comment:
I deliberately do not allow input to be `str`, for there might be encoding issue ('utf-8'?) here, and the interface of `set` would be not uniform. If one would like to pass in knowledge contained in `str`, one has to convert to `bytes`.